src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.Unix.cs

```diff
@@ -19,6 +19,8 @@ public partial class Ping
         private const int IcmpHeaderLengthInBytes = 8;
         private const int MinIpHeaderLengthInBytes = 20;
         private const int MaxIpHeaderLengthInBytes = 60;
+        private static bool _sendIpHeader = RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
```

Comment:
```suggestion
        private static readonly bool _sendIpHeader = RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
```
Same below for `_needsConnect`.